jira/utils/__init__.py

```diff
@@ -79,3 +82,79 @@ def json_loads(resp: Response | None) -> Any:
         if not resp.text:
             return {}
         raise
+
+
+def cloud(client_method: Callable) -> Callable:
+    """A convenience decorator to check if the Jira instance is cloud.
+
+    Checks if the client instance is talking to Cloud Jira. If it is, return
+    the result of the called client method. If not, return None and log a
+    warning.
+
+    Args:
+      client_method: The method that is being called by the client.
+
+    Returns:
+      Either the result of the wrapped function or None.
+    """
+
+    @functools.wraps(client_method)
+    def check_if_cloud(*args, **kwargs):
+        # NOTE(jpavlav): The first argument of any class instance is a `self`
```

Comment:
On this point, you may want to check if the first argument is an instance of the JIRA class. As otherwise this decorator might be used incorrectly

Reply:
Yeah, I considered that, but thought that the utilization was pretty clear, and that if there was a failure, it would be readily apparent why. Having said that, I can go ahead and make that adjustment. I think the only other concern I had was circular import dependency. Since `jira.client` depends on `jira.utils`, but I think it would only matter if something from `jira.utils` was actually a member of one of the classes in `client`. Either way, will see what I can do.

Reply:
Yeah, what I was afraid of here is the case. I can't check if the `self` argument `isinstance` of the `JIRA` class because that would necessitate importing it and causes a circular dependency:

```python
ImportError: cannot import name 'JIRA' from partially initialized module 'jira.client' (most likely due to a circular import) (/Users/jpalmer/Repositories/github.com/Personal/jira/jira/client.py)
```

I guess I'll refactor this a bit. I'll put these decorators in the `client`. I don't think it's unreasonable for them to exist there, they pertain to the client specifically so it should be fine.